ctapipe/image/muon/muon_integrator.py

```diff
@@ -240,25 +243,35 @@ def image_prediction(self, impact_parameter, phi, centre_x,
         # Produce smoothed muon profile
 
         ang_prof, profile = self.plot_pos(impact_parameter, radius, phi)
-        # Produce gaussian weight for each pixel give ring width
+        # Produce gaussian weight for each pixel given ring width
         radial_dist = np.sqrt((pixel_x - centre_x)**2 + (pixel_y - centre_y)**2)
-        gauss = norm.pdf(radial_dist, radius, ring_width)
+        # The weight is the integral of the ring's radial gaussian profile inside the ring's width
+        gauss = norm.cdf(radial_dist+0.5*self.pixel_width.value, radius, ring_width) - \
+                norm.cdf(radial_dist-0.5*self.pixel_width.value, radius, ring_width)
 
         # interpolate profile to find prediction for each pixel
         pred = np.interp(ang, ang_prof, profile) * u.m
 
-        # Multiply by integrated emissivity between 300 and 600 nm
-        photval = self.photemit / u.deg
+        # Multiply by integrated emissivity between 300 and 600 nm, and rest of factors to get total number of photons per pixel
+        photval = self.photemit # would be per radian, but no need to put it here, would anyway cancel out below
         pred *= 0.5 * photval
+        pred *= (self.pixel_width.value / radius) # multiply by angle (in radians) subtended by pixel width as seen from ring center
+        pred *= np.sin(2 * (radius*u.deg).to_value(u.rad))
 
-        # weight by pixel width
-        pred *= (self.pixel_width.value / radius)
-        pred *= np.sin(2 * radius)
-        # weight by gaussian width
-        pred *= self.pixel_width.value * gauss
+        # multiply by gaussian weight, to account for "fraction of muon ring" which falls within the pixel
+        pred *= gauss
+
+        # Now it would be the total light in an area S delimited by: two radii of the ring, tangent to the sides of the pixel in question,
+        # and two circles concentric with the ring, also tangent to the sides of the pixel.
+        # A rough correction, assuming pixel is round, is introduced here: [pi*(pixel_width/2)**2]/ S. Actually, for the large rings (relative to
+        # pixel size) we are concerned with, a good enough approximation is the ratio between a circle's area and that of the square whose side
+        # is equal to the circle's diameter. In any case, since in the end we do a data-MC comparison of the muon ting analysis outputs, it is not
+        # critical that this value is exact.
+        pred *= 0.7854
```

Comment:
I guess it's np.pi / 4, so the ratio of the areas of unit circle and square.

Please replace by definining a constant `CIRCLE_SQUARE_AREA_RATIO = np.pi / 4`

Reply:
It is just what max said, and hence a rough approximation for hexagonal pixels of any size.
Could be done with the pixel area and the area of the "ring sector" in which the pixel is contained, but it would not be 100% exact either - pixel orientation introduces differences. In any case this is irrelevant, since in the end one compares data and MC with same camera geometry. I would be for just introducing Max's suggestion.

Reply:
> Does this come from the formula in the comment? 
The formula would be _not_ for a square around the circle, but for a "ring sector" around the circle.  For muon rings of ~1 deg radius and pixels O(0.1 deg) the difference is negligible.

Reply:
Ok that makes more sense - Then just replace the bare constant as @MaxNoe suggests (we explicitly forbid bare constants in the coding guidelines). Perhaps update the comment to be more clear what is actually done.